hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordReader.java

```diff
@@ -81,11 +80,16 @@ private HoodieMergedLogRecordReader(HoodieReaderContext<T> readerContext,
                                       Option<String> partitionName,
                                       InternalSchema internalSchema,
                                       Option<String> keyFieldOverride,
-                                      boolean enableOptimizedLogBlocksScan, HoodieRecordMerger recordMerger) {
+                                      boolean enableOptimizedLogBlocksScan,
+                                      HoodieRecordMerger recordMerger,
+                                      HoodieFileGroupRecordBuffer<T, K> recordBuffer) {
     super(readerContext, fs, basePath, logFilePaths, readerSchema, latestInstantTime, readBlocksLazily, reverseReader, bufferSize,
-        instantRange, withOperationField, forceFullScan, partitionName, internalSchema, keyFieldOverride, enableOptimizedLogBlocksScan, recordMerger);
+        instantRange, withOperationField, forceFullScan, partitionName, internalSchema, keyFieldOverride, enableOptimizedLogBlocksScan,
+        recordMerger, recordBuffer);
     this.records = new HashMap<>();
     this.scannedPrefixes = new HashSet<>();
+    this.recordBuffer = recordBuffer;
```

Comment:
Do we need to assign the record buffer 2 times?

Reply:
Removed it.